jina/serve/runtimes/worker/http_gcp_app.py

```diff
@@ -0,0 +1,199 @@
+
+    from jina.proto import jina_pb2
+    from jina.serve.runtimes.gateway.models import _to_camel_case
+
+    if not docarray_v2:
+        logger.warning('Only docarray v2 is supported with GCP. ')
+        return
+
+    class Header(BaseModel):
+        request_id: Optional[str] = Field(
+            description='Request ID', example=os.urandom(16).hex()
+        )
+
+        class Config(BaseConfig):
+            alias_generator = _to_camel_case
+            allow_population_by_field_name = True
+
+    class InnerConfig(BaseConfig):
+        alias_generator = _to_camel_case
+        allow_population_by_field_name = True
+
+    class VertexAIResponse(BaseModel):
+        predictions: Any = Field(
+            description='Prediction results',
+        )
+
+    app = FastAPI()
+
+    if cors:
+        app.add_middleware(
+            CORSMiddleware,
+            allow_origins=['*'],
+            allow_credentials=True,
+            allow_methods=['*'],
+            allow_headers=['*'],
+        )
+        logger.warning('CORS is enabled. This service is accessible from any website!')
+
+    def add_post_route(
+        endpoint_path,
+        input_model,
+        output_model,
+        input_doc_list_model=None,
+    ):
+        from docarray.base_doc.docarray_response import DocArrayResponse
+
+        app_kwargs = dict(
+            path=f'/{endpoint_path.strip("/")}',
+            methods=['POST'],
+            summary=f'Endpoint {endpoint_path}',
+            response_model=VertexAIResponse,
+            response_class=DocArrayResponse,
+        )
+
+        def is_valid_csv(content: str) -> bool:
+            import csv
+            from io import StringIO
+
+            try:
+                f = StringIO(content)
+                reader = csv.DictReader(f)
+                for _ in reader:
+                    pass
+
+                return True
+            except Exception:
+                return False
+
+        async def process(body) -> output_model:
+            req = DataRequest()
+            if body.header is not None:
+                req.header.request_id = body.header.request_id
+
+            if body.parameters is not None:
+                req.parameters = body.parameters
+            req.header.exec_endpoint = endpoint_path
+            req.document_array_cls = DocList[input_doc_model]
+
+            data = body.data
+            if isinstance(data, list):
+                req.data.docs = DocList[input_doc_list_model](data)
+            else:
+                req.data.docs = DocList[input_doc_list_model]([data])
+                if body.header is None:
+                    req.header.request_id = req.docs[0].id
+
+            resp = await caller(req)
+            status = resp.header.status
+
+            if status.code == jina_pb2.StatusProto.ERROR:
+                raise HTTPException(status_code=499, detail=status.description)
+            else:
+                return VertexAIResponse(predictions=req.data.docs)
```

Comment:
I wonder why I got this error when running `pytest -v -s -x test_gcp.py -k test_provider_gcp_deployment_inference`:

```
  File "pydantic/main.py", line 449, in pydantic.main.BaseModel.dict
  File "pydantic/main.py", line 868, in _iter
  File "pydantic/main.py", line 794, in pydantic.main.BaseModel._get_value
  File "/Users/zacli/Code/jina/env/lib/python3.9/site-packages/docarray/array/doc_list/doc_list.py", line 135, in __init__
    super().__init__(docs)
  File "/Users/zacli/Code/jina/env/lib/python3.9/site-packages/docarray/array/doc_list/doc_list.py", line 163, in _validate_docs
    yield self._validate_one_doc(doc)
  File "/Users/zacli/Code/jina/env/lib/python3.9/site-packages/docarray/array/doc_list/doc_list.py", line 170, in _validate_one_doc
    raise ValueError(f'{doc} is not a {self.doc_type}')
ValueError: {'id': '5d814334ba65cd66b356f363262dd2ad', 'text': 'hello world', 'embeddings': NdArray([[0.72778929, 0.23900942, 0.03286786, 0.95298338, 0.77614308,
          0.03805365, 0.52810364, 0.70904448, 0.03221386, 0.40657516,
          0.05665328, 0.73416605, 0.04400287, 0.79719694, 0.42839442,
          0.43580189, 0.07334091, 0.60751732, 0.10342848, 0.51660762,
          0.06586024, 0.88129511, 0.42874586, 0.88633873, 0.31858046,
          0.71453457, 0.31093969, 0.92729799, 0.25257687, 0.92740828,
          0.61681301, 0.51788409, 0.00997884, 0.21210052, 0.91713272,
          0.37283237, 0.53357638, 0.77158797, 0.96699236, 0.13919022,
          0.92879329, 0.78211408, 0.65285461, 0.89399441, 0.42854324,
          0.94649221, 0.1966403 , 0.39643899, 0.42736609, 0.73116549,
          0.88923411, 0.9894739 , 0.9124282 , 0.83614148, 0.27662465,
          0.69740887, 0.34939623, 0.07094385, 0.38615401, 0.04148779,
          0.0133029 , 0.68877819, 0.60318039, 0.64945002]])} is not a <class 'executor.EmbeddingResponseModel'>
```